meson.build

```
@@ -0,0 +1,36 @@
+project('tlmi-auth', 'c',
+    version : '1.0.1',
+    license : 'GPL-2.0-or-later',
+    default_options : [
+        'buildtype=debugoptimized',
+        'b_asneeded=True',
+        'b_lundef=True',
+        'b_pie=true',
+        'c_std=gnu11',
+        'warning_level=3',
+    ],
+    meson_version : '>=0.54.0')
```

Any particular reason to pin a relatively high minimum version? The current requirement isn't available in e.g. Debian buster (oldstable) or Ubuntu 20.04.

It compiles fine (without any warnings) on pretty old versions of Meson -- I only tested 0.45.0, because that is what is available in Ubuntu 18.04 in addition to both other distro releases.

Now, granted, it's pretty easy to pip install a suitable version. :) But there's no harm in allowing the distro-available version to be used, AFAICT.

```suggestion
    meson_version : '>=0.45.0')
```

---

Three reasons:

1. That's the lowest I was willing to test (RHEL 8)
2. You can't add new packages to those old distributions normally.
3. `meson compile` was added in 0.54. :)

---

Also, I'm not sure if updated ThinkLMI is present in their kernels. I know it exists in RHEL and SLE (both of which have updated Meson versions) and obviously Fedora/Arch/openSUSE Tumbleweed, etc. But otherwise... 🤷🏾‍♂️

---

Well, I just did a `pip install --user` for testing purposes. :D But also, lowering the meson_version to a very low number produced no "feature too new for your meson_version" warnings, which is usually a good sign that things work on those low versions.

I don't think `meson compile` is very needed, anyway, since that's just a portability script for:
- forcing the Windows MSVC compiler to be added to the environment before ninja is run (due to a ninja bug)
- being agnostic to whether ninja (the default on all platforms) or vs / xcodebuild (optional alternative backends only available for Windows and macOS) get run. I think in most cases everyone wants to use ninja anyway. ;) Regardless, it doesn't matter to meson_version, which version of the additional meson utilities you use.

---

I don't trust Meson's ability to be compatible enough for that. 😛 

Been burned too much for that.

---

> Also, I'm not sure if updated ThinkLMI is present in their kernels.

Yeah, that's often pretty hard to tell because Ubuntu has a habit of backporting major features, patch by patch, into old kernels while keeping the old version number "for stability". :rofl:

---

I should know the answer to the Ubuntu kernel version but I honestly can't remember. It didn't get backported for quite a while but did eventually get into their OEM kernel. If it's important I can find out.

Not sure if it's important - but the tlmi_crypto piece of this does work for Windows. For a little while I (internal to Lenovo) had this project setup so tlmi_crypto could be used with Windows as well and just have a different front end; but the Windows team weren't interested in sharing so I got a little bit annoyed and...erm.....ripped out all the pieces I'd done for them, and went Linux only.

As it stands it's use is Linux only and there is no particular plan to change that.